### ISSUE TITLE
Improve the accessibility of note History dialog

### ISSUE THREAD

**Issue by pachlava** [Automattic/simplenote-electron#2898]
### What
When the history dialog is opened, and the note is big enough, the dialog will cover a part of the note in this way:

https://user-images.githubusercontent.com/73365754/117119256-771ead00-ad9a-11eb-8721-16e77167551a.mov

To see the changes in the covered part, the only available method is scrolling. In a case when scrolling is impossible, there is no possibility to see the changes in this part of the note, because the dialog is closed on click-out, and because the focus is trapped inside the dialog.

### Why
Improve the accessibility of the app.

### [How](p1620139812028900-slack-C01KY08KQSY)

> 1. A small win would be to disable the click outside from closing the history slider, but that introduces other oddities we’d need to address as well like you wouldn’t be able to click on another note in the note list or any menu items until you cancel the note history, but it looks like you should be able to.

> 2. Currently, when the revision slider value changes, the revision date is announced. We could announce additional information if it is deemed helpful. It should likely be explored/debated whether announcing the entire selected note revision or a "diff" of the revision is a helpful announcement. Either one could potentially be a lot of text and overwhelming. However, that may be the "norm" or expected as a screen reader user — I'm not sure. We'd likely need to build it and play with it to get a sense for how helpful it is.

> 3. As you mentioned, we could stop hiding the Note Editor UI and allow the user to navigate into that UI while the Revision Selector is open. This would allow the user to "read" the entire note, and scroll if needed. We would need to figure out a way to focus trap the Note Editor + Revision Selector, which may require restructuring HTML or additional JavaScript.

> 4. If we ultimately allow navigating into the Note Editor, we likely should no longer declare the Revision Selector a dialog, but something else (i.e. change role="dialog"). Dialogs are generally defined as UI that overlays all other UI preventing access to anything else.

**Comment by pachlava** [comment]
> Human1: My two cents - I think from a user perspective, being able to scroll the note while the history panel is open is ideal. I would suggest that while the history is visible we disable editing of the note and allow scrolling. Navigating anywhere else should immediately dismiss the history panel without restoring. So essentially, we can listen for clicks on the document and if the target is not the history or editor then dismiss the history panel.

> Human2: Do you mean that clicking (or tabbing) to note editor from history should not dismiss history, and allow to scroll with keys (along with usual scroll ofc)?

> Human1: Yeah, the editor would be a preview exactly as you said and would be scrollable when the History UI is shown with the History panel being fixed position.
